specs/disk_details_for_hardware_inventory.adoc

```diff
@@ -0,0 +1,154 @@
+  details and update those details into node_inventory list.
+* Create a new class in persistence module to keep disk details.
+* Add a new function in persister to persist disk details.
+* Retrieve inventory details from tendrl-node-inventory file and compare it with
+  node_inventory list, If any changes then delete the disk details from etcd and
+  update with newly collected disk details.
+* Modify the manager to populate the instance of newly created class using
+  node_inventory list and persist the instance in etcd.
+* Store free disk details with different key and used disk details with
+  different key.
+* Create an object in tendrl_definitions_node_agent for disk details.
+
+== Alternatives
+
+None
+
+== Data model impact
+
+* Add used disk details in etcd with key nodes/node_id/Disks/used/disk_id.
+* Add free disk details in etcd with key nodes/node_id/Disks/free/disk_name.
+* Add disk object in tendrl_definitions_node_agent/disk.
+
+== Impacted modules
+
+== Tendrl API impact:
+
+None
+
+=== Tendrl/common impact
+
+None
+
+=== Tendrl/node_agent impact
+
+* Add a new function in pull_hardware_inventory to collect disk details.
+* Create a new class in persistence module to keep the disk details.
+* Add a new function in persister to persist disk details.
+* Retrieve inventory details from tendrl-node-inventory file and compare it with
+  node_inventory list, If any changes then delete the disk details from etcd and
+  update with newly collected disk details.
+* Modify the manager modules to read and store disk details.
+* Store free disk details with different key and used disk details with
+  different key.
+* Create an object in tendrl_definitions_node_agent for disk details.
+* Commands used to retrieve disk details are:
+```
+columns = 'NAME,KNAME,FSTYPE,MOUNTPOINT,UUID,PARTUUID,MODEL,SIZE,TYPE,PKNAME,
+VENDOR'
+lsblk = ("lsblk --all --bytes --noheadings --output='%s' --path --raw" % columns)
+```
+* Excpected output format for the disk details is:
+```
+{[{"device_name":   "devicename",
+       "fs_type":     "fstype",
+       "fs_uuid":     "uuid",
+       "model":      "model",
+       "mount_point": ["mountpoint", ...],
+       "name":       "name",
+       "parent":     "parentdevicename",
+       "size":       uint64,
+       "disk_type":  "type",
+       "used":       boolean,
+       "ssd":        boolean,
+       "vendor":     "string",
+       "disk_id":     "uuid"}, ...], ...}
+```
+
+=== Tendrl/gluster_integration impact
+
+None
+
+=== Tendrl/ceph_integration impact
+
+None
+
+
+== Security impact
+
+None
+
+== Notifications/Monitoring impact
+
+None
+
+
+== Other end user impact
+
+None
+
+== Performance Impact
+
+None
+
+== Other deployer impact
+
+None
+
+== Developer impact
+
+None
+
+== Implementation
+
+https://github.com/Tendrl/node_agent/issues/78
+
+== Assignee(s)
+
+Primary assignee: Gowtham Shanmugasundaram
+
+Other contributor(s): TODO
+
+== Work Items
+
+* Add Util-linux as a dependency.
```

Comment:
@r0h4n i am using lsblk command for find disk details.. do we really need this package... It is a default pakcage in linux right?